### ISSUE TITLE
What happens if we already have BEARER_TOKEN_FILE set to an invalid token?

### ISSUE THREAD

**Issue by shreyb** [fermitools/jobsub_lite#497]
@cathulhu asked a good question yesterday while we were discussing job submission and tokens:  If we have an invalid token (whether expired or for the wrong experiment), and try to submit a job, how does jobsub_lite handle it?

If `BEARER_TOKEN_FILE` is not set in the environment, we should be fine, because jobsub_lite should ignore the bad credential and get the correct token.  However, if it **is** set, do we just crash and burn, or give a helpful error message?  We should investigate this, and if needed, make sure we handle that use case gracefully.

I think the correct behavior should be that if `BEARER_TOKEN_FILE` is set to an invalid credential:

1.  We fail the submission.
2. Do not get a new token.
3. Let the user know WHY the submission failed, 
4. Advising the user to either:
    a. Unset `BEARER_TOKEN_FILE`, or 
    b. Get a new token into the location `BEARER_TOKEN_FILE` is set to.

For (3) above, I don't think we'd need to give a concrete reason besides the fact that `BEARER_TOKEN_FILE` is set to an invalid token.

@marcmengel - would you agree that this is the correct behavior or do you have another suggestion?

**Comment by shreyb** [comment]
Did some investigation, and we do indeed need to fix this.  Here were my tests (using the 1.5 release):

## Using an invalid file for `BEARER_TOKEN_FILE`

$ echo "thisisabadfile" > /tmp/testbadtoken
$ export BEARER_TOKEN_FILE=/tmp/testbadtoken
$ jobsub_submit -G fermilab file:///usr/bin/sleep 300


Error: InvalidTokenFormat: Serialized token is not a readable format.
```

This is OK, but we should probably give a better explanation of what's going on there.

## Using a valid token, but for the wrong VO

$ htgettoken -a htvaultprod.fnal.gov -i dune
...
Storing bearer token in /tmp/testbadtoken
$ jobsub_submit -G fermilab file:///usr/bin/sleep 300
...
Submitting job(s).
1 job(s) submitted to cluster 65770261.
```

OK, so job submission succeeded.  That's somewhat expected.  Here's what's bad.  In the JDF (.cmd file), we see the following lines:

```
use_oauth_services = fermilab

fermilab_oauth_permissions_6e4d19314c = " compute.read compute.create compute.cancel compute.modify storage.read:/dune storage.create:/fermigrid/jobsub/jobs ..."
```

So the job gets submitted, but the requested scopes are wrong, which could have bad downstream effects, since now the token being ostensibly handed to the job has the wrong scopes.

We should be doing some sort of check here, and failing the job submission, IMO.

**Comment by shreyb** [comment]
This work is done.  Here are the new tests:

## Using an invalid file for BEARER_TOKEN_FILE

```
$ echo "thisisabadfile" > /tmp/testbadtoken
$ export BEARER_TOKEN_FILE=/tmp/testbadtoken
$ jobsub_submit -G fermilab file:///usr/bin/sleep 300

Error: InvalidTokenFormat: Token stored in $BEARER_TOKEN_FILE is not in a readable format. Please inspect the token with httokendecode or unset $BEARER_TOKEN_FILE and allow jobsub to create a new token.
```

Same as before, but more helpful.  Good.

## Using a valid token, but for the wrong VO

This was the big problem in 1.5.  Here's what happens now:

```
$ htgettoken -a htvaultprod.fnal.gov -i dune
...
Storing bearer token in /tmp/testbadtoken
$ ~/jobsub_lite/bin/jobsub_submit -G fermilab file:///usr/bin/sleep 300


Error: ValueError: BEARER_TOKEN_FILE contains a token with the wrong group or role. jobsub expects a token with group fermilab and role Analysis. Instead, BEARER_TOKEN_FILE contains a token with group dune and role Analysis.
```

Much better.

**Comment by shreyb** [comment]
Fixed by #509
